x-pack/plugins/elastic_assistant/server/routes/post_actions_connector_execute.ts

```diff
@@ -158,6 +172,58 @@ export const postActionsConnectorExecuteRoute = (
               });
             }
 
+            if (conversation?.title === 'New chat' && prevMessages) {
+              try {
+                const autoTitle = await executeAction({
+                  actions,
+                  request,
+                  connectorId,
+                  actionTypeId: connectors[0]?.actionTypeId,
+                  params: {
+                    subAction: request.body.subAction,
+                    subActionParams: {
+                      model: request.body.model,
+                      messages: [
+                        {
+                          role: 'assistant',
+                          content:
+                            'You are a helpful assistant for Elastic Security. Assume the following message is the start of a conversation between you and a user; give this conversation a title based on the content below. DO NOT UNDER ANY CIRCUMSTANCES wrap this title in single or double quotes. This title is shown in a list of conversations to the user, so title it for the user, not for you.',
+                        },
+                        newMessage ?? prevMessages?.[0],
+                      ],
+                      ...(connectors[0]?.actionTypeId === '.gen-ai'
+                        ? { n: 1, stop: null, temperature: 0.2 }
+                        : { temperature: 0, stopSequences: [] }),
+                    },
+                  },
+                  logger,
+                });
+                if ((autoTitle as StaticResponse).status === 'ok') {
```

Comment:
Would you be willing to move the type cast to the const `autoTitle` so we do not have to repeat?
```suggestion
                if (autoTitle.status === 'ok') {
```

Reply:
Also changing `subAction: request.body.subAction` to `subAction: 'invokeAI'` as we don't want to accidentally use streaming which will return a `Readable`. Adding note to update `executeAction` to use function overloading to prevent this cast.